src/box/sql/build.c

```diff
@@ -1194,85 +1196,28 @@ vdbe_emit_fk_constraint_create(struct Parse *parse_context,
 	bool is_alter_add_constr =
 		parse_context->create_table_def.new_space == NULL &&
 		parse_context->create_column_def.space == NULL;
-	if (!is_alter_add_constr) {
-		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->child_id,
-				  constr_tuple_reg + 1);
-	} else {
-		sqlVdbeAddOp2(vdbe, OP_Integer, fk->child_id,
-				  constr_tuple_reg + 1);
-	}
-	if (!is_alter_add_constr && fk_constraint_is_self_referenced(fk)) {
-		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->parent_id,
-				  constr_tuple_reg + 2);
+	if (!is_alter_add_constr)
+		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->child_id, regs);
+	else
+		sqlVdbeAddOp2(vdbe, OP_Integer, fk->child_id, regs);
+	if (!is_alter_add_constr && fk_constraint_is_self_referenced(fk))
+		sqlVdbeAddOp2(vdbe, OP_SCopy, fk->parent_id, regs + 1);
+	else
+		sqlVdbeAddOp2(vdbe, OP_Integer, fk->parent_id, regs + 1);
+	if (fk->is_field_fk) {
+		sqlVdbeAddOp2(vdbe, OP_Integer, fk->links->child_field,
+			      regs + 2);
+		sqlVdbeAddOp2(vdbe, OP_Integer, fk->links->parent_field,
+			      regs + 3);
 	} else {
-		sqlVdbeAddOp2(vdbe, OP_Integer, fk->parent_id,
-				  constr_tuple_reg + 2);
+		uint32_t size;
+		char *raw = fk_constraint_encode_links(fk, &size);
```

Comment:
If `raw` is `NULL`, we probably have to set `parse_context->is_aborted`.

Btw, I also see that all these `sqlVdbeAddOp...()` don't set `->is_aborted` too. They should do it in `growOpArray()` func. Can also leave it as is if we will port it to `xmalloc()` soon anyway.

Reply:
Thank you for the review! Fixed.